mlibrary/lib.umich.edu
# Website 78 replace moment dependency

# Overview
Moment.js is no longer supported and was built on an old era of the JS ecosystem. Replacement options included date-fns _[[website](https://date-fns.org/), [github](https://github.com/date-fns/date-fns)]_ and day.js _[[website](https://day.js.org/), [github](https://github.com/iamkun/dayjs)]_. I ultimately decided to go with date-fns since it appears to have a larger user base and more support.

This PR should move everything from moment to date-fns. I'm not 100% sure what the locale formatting of parseISO is doing despite my best efforts to understand it. My goal was to create the same outputs from moment with date-fns and I believe I've done that.

All changed components:
- src/components/panels/events-and-exhibits-panel.js
- src/components/panels/featured-and-latest-news.js
- src/components/panels/hours-lite-panel.js
- src/components/panels/hours-panel.js
- src/components/todays-hours.js
- src/templates/news-landing.js
- src/templates/event.js
- src/templates/news.js
- src/utils/events.js
- src/utils/hours.js

[date-fns documentation](https://date-fns.org/docs/Getting-Started)

> This pull request resolves/closes/fixes [WEBSITE-78](https://mlit.atlassian.net/browse/WEBSITE-78).

Let's remove moment with the June website updates.

## Anything else?
[All conversion tracking was done in this document.](https://docs.google.com/spreadsheets/d/1F9PTjNtmXse1CDv4SM8_KAIWGRv4ezkeD-7yNa8D8MQ/edit#gid=289530058) I linked the second tab (Pre Deploy Testing Status (Phase 2)) where I share some additional notes about the changes and the live site with an example of where that component appears. The first tab is my development tracking document for myself.

## Testing
List instructions on how to test the pull request. Some examples:

- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
